.github/workflows/ci.yaml

```diff
@@ -476,7 +476,7 @@ jobs:
         with:
           path: venv
           key:
-            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{
+            ${{ runner.os }}-${{ matrix.python-version }}-${{
```

Comment:
This shouldn't be necessary. The `Set up Python` version will output `${{ matrix.python-version }}`. As far as I can tell, the tests didn't crash.

Reply:
Hmm, this doesn't work as well.

Please see the version of `pytest` in the `pypy-3.7` run. That's the `3.6` version. You can see they are using the same cache in the `Restore Python virtual environment` job. The `key` is the same.

However, my fix also doesn't work. I guess something weird is going on with `python-version` and `PyPy`? For the `cpython` jobs the keys do work..

Reply:
This pretty much looks like a bug in the `setup-python` actions. If I had to guess a result of https://github.com/actions/setup-python/pull/342.

Reply:
I've included the fix in #5868
Not sure when the bug will be addressed. We can always revert it, once a new version is released.

After #5868 is done, we should be able to move forward here.